src/js/builtins/BundlerPlugin.ts

```diff
@@ -105,6 +106,38 @@ export function loadAndResolvePluginsForServe(
   return promiseResult;
 }
 
+export function runOnEndCallbacks(
+  this: BundlerPlugin,
+  buildResult: Bun.BuildOutput,
+  buildPromise: Promise<Bun.BuildOutput>,
+): Promise<void> | void {
+  if (!this.onEndCallbacks) return;
+  const promises: PromiseLike<unknown>[] = [];
+
+  for (const callback of this.onEndCallbacks) {
+    try {
+      const result = callback(buildResult);
+
+      if (result && $isPromise(result)) {
+        $arrayPush(promises, result);
+      }
+    } catch (e) {
+      $arrayPush(promises, Promise.reject(e));
+    }
+  }
+
+  if (promises.length > 0) {
+    return Promise.all(promises).then(
+      () => {
+        $resolvePromise(buildPromise, buildResult);
+      },
+      e => {
+        $rejectPromise(buildPromise, e);
+      },
+    ); // return this promise - it gets "awaited" in the bundler before the whole build ends
```

Comment:
it does not get awaited, we just don't do anything with it

Reply:
it was a leftover comment from previous attempt, will remove. thx